jdbc/src/main/java/org/apache/metamodel/jdbc/JdbcDataSet.java

```diff
@@ -197,8 +197,12 @@ public void close() {
         if (_closed) {
             return;
         }
+
+        FileHelper.safeClose(_resultSet);
+
         if (_jdbcDataContext != null) {
-            _jdbcDataContext.close(_connection, _resultSet, _statement);
+            FileHelper.safeClose(_statement);
```

Comment:
Shouldn't this be placed outside of the `if`, just like the closing of the ResultSet above?